### ISSUE TITLE
Bookmarks panel UX is terrible

### ISSUE THREAD

**Issue by shamazmazum** [atlas-engineer/nyxt#3433]
**Purpose of request:**
Improve `bookmarks-panel` 

**Describe your proposed change:**
Somewhere in the past I did grouping of bookmarks shown by `bookmarks-panel` by tags. This command showed a list of tags, so you could expand any tag and view all bookmarks tagged with this tag. Now it is changed so all groups are already expanded and you have to collapse each group to see the whole list of tags. This makes navigation harder, IMHO.

Moreover, if you click at any of the bookmarks, all groups become open again, and you have to start all over again collapsing them and searching for a needed tag.

This is how I fixed it for myself:

* Starting with all groups collapsed: https://github.com/shamazmazum/nyxt/commit/d6b5aebfc083213f15236e1206caa22984aab19c
* Not collapsing them again on every opened bookmark: https://github.com/shamazmazum/nyxt/commit/a392d89be55b9d0f4ea871eaab09e49c22e7fd41

**Why do you believe this to be an improvement?**
Clearer way to navigate through bookmarks

---

### Pre-Build Checklist:
- [ ] Discussion has occurred.
- [ ] One or more developers have signed off on this change.
- [ ] One or more user researchers have signed off on this change.

**Comment by aadcg** [comment]
@shamazmazum I agree that the bookmarks panel UX is subpar, but I'm not sure if the issues you mention are the most critical.

We have two commands that list bookmarks: `list-bookmarks` and `bookmarks-panel`. As the name suggest, one uses a buffer while the other uses a panel buffer. Notice that they use different defaults when it comes to expanding each bookmark tag. Why so? Because in the main buffer (unlike in the panel buffer) you can invoke `follow-hint` to expand/collapse tags or make use of `{next,previous}-heading` and `jump-to-heading`.

In short, while it's trivial to traverse the bookmark structure in the main buffer with the keyboard alone, it is hard to do the same in the panel buffer.  With that in mind, it seems reasonable that the default UI for the panel buffer resembles a dumb item list.

Interested in your thoughts.

**Comment by shamazmazum** [comment]
As you cannot use `follow-hint` with a panel, it seems that you have to use a mouse to interact with it anyway, don't you? So why not to make this interaction a bit more pleasant? :) For me, to find a needed bookmark is to collapse all the groups, find a needed group and then find the bookmark in that group. Before collapsing, it's more like a plain list to me. Not sure, how other people do a search in their bookmarks, but my workflow is the one I described.

Also, expanding them again after I found the needed bookmarks feels awful. What if you want to open another bookmark from that group?

**Comment by shamazmazum** [comment]
I think, yes. E.g., you can create a slot in `panel-page` which tells "a buffer which shows that page must be reloaded when you do something with other buffers". Then the value in this slot propagates to `panel-buffer` and any buffer operation checks that value.

Here is a draft: https://github.com/shamazmazum/nyxt/commit/18e30227f60b2631e5b504511c991e22cec6f942

I think, I won't change `define-panel-command(-global)` signature in the final commit.

**Comment by aadcg** [comment]
Hmm, I had another in mind. Why not tweak the condition on `buffers-panel-handler-set-buffer` and `buffers-panel-reload-handler` such that it is triggered on `buffers-panel`s?

**Comment by shamazmazum** [comment]
Thanks for the fix!
